packs/octopusdeploy/sensors/new_deployment_sensor.py

```diff
@@ -0,0 +1,136 @@
+import time
+
+from lib.sensors import OctopusDeploySensor
+
+
+class NewDeploymentSensor(OctopusDeploySensor):
+    """
+    * self._sensor_service
+        - provides utilities like
+            get_logger() for writing to logs.
+            dispatch() for dispatching triggers into the system.
+    * self._config
+        - contains configuration that was specified as
+          config.yaml in the pack.
+    * self._poll_interval
+        - indicates the interval between two successive poll() calls.
+    """
+
+    def __init__(self, sensor_service, config=None, poll_interval=None):
+        super(NewDeploymentSensor, self).__init__(sensor_service=sensor_service,
+                                                  config=config,
+                                                  poll_interval=poll_interval)
+        self._trigger_ref = 'octopusdeploy.new_deployment'
+        self._logger = self._sensor_service.get_logger(__name__)
+        self._deploy_key = 'octopusdeploy.last_deploy_date_str'
+        self._last_date = None
+
+    def setup(self):
+        pass
+
+    def poll(self):
+        self._logger.debug('Requesting list of deployments')
+        deployments = self._get_deployments()
```

Comment:
Does API support for kind of filtering (based on a date or similar), or do you always need to retrieve all the available deployments?